bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/ReadmeHelper.java

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.miio.internal;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.openhab.binding.miio.internal.basic.MiIoBasicChannel;
+import org.openhab.binding.miio.internal.basic.MiIoBasicDevice;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Support creation of the miio readme doc
+ *
+ * Run after adding devices or changing database entries of basic devices
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+public class ReadmeHelper {
```

Comment:
This fill is better to be in the `src/test/java` directory as it's not part of the actual binding. With the maven/bnd build system we can have that structure now. You could even make it a real test. Maybe add `@Ignore` or add a check if the README.md exists and if not don't run the test. Then simply by removing the README.md and run the tests a new README can generated. 
There is also a discussion https://github.com/openhab/openhab-addons/issues/6545 on generating the tables from the xml files. For this it's somewhat different. But there we also need to put some markers in the readme to inline the generated tables. That same process might also be useful here as with the current implementation if someone would simply make a pr on the readme it can get easily lost if the base readme isn't updated.

Reply:
I've now moved it to the test folder indeed. 
Added the ignore as well as indeed changes to the regular readme file get lost easilly.
That way, at least we need to see it the change to the readme in a PR before the change is committed.